src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -3,34 +3,88 @@ import axios from "axios";
+	errorCode: string;
+	message: string;
+	statusCode: number;
+}
+
+interface GheServerUrlErrorResponses {
+	[key: string | number]: MessageAndCode;
+}
+
+interface GheServerUrlErrors {
+	codeOrStatus: GheServerUrlErrorResponses;
+}
+
+const gheServerUrlErrors: GheServerUrlErrors = {
+	codeOrStatus: {
+		invalidUrl: {
+			errorCode: "GHE_ERROR_INVALID_URL",
+			message: "Invalid URL",
+			statusCode: 200
+		},
+		ENOTFOUND: {
+			errorCode: "GHE_ERROR_ENOTFOUND",
+			message: "Request to URL failed",
+			statusCode: 200
+		},
+		502: {
+			errorCode: "GHE_SERVER_BAD_GATEWAY",
+			message: "Bad gateway",
+			statusCode: 502
+		},
+		default: {
+			errorCode: "GHE_ERROR_DEFAULT",
+			message: "Something went wrong",
+			statusCode: 200
+		}
+	}
+};
+
+export const getGheErrorMessages = (codeOrStatus: number | string | null) => {
+	switch (codeOrStatus) {
+		case "invalidUrl":
+			return gheServerUrlErrors.codeOrStatus["invalidUrl"];
+		case "ENOTFOUND":
+			return gheServerUrlErrors.codeOrStatus["ENOTFOUND"];
+		case 502:
+			return gheServerUrlErrors.codeOrStatus[502];
+		default:
+			return gheServerUrlErrors.codeOrStatus["default"];
+	}
+};
+
 export const JiraServerUrlPost = async (
 	req: Request,
 	res: Response
 ): Promise<void> => {
-	const { gheServerURL, installationId } = req.body;
+	const { gheServerURL } = req.body;
+	const { id: installationId } = res.locals.installation;
 
 	req.log.debug(`Verifying provided GHE server url ${gheServerURL} is a valid URL`);
 	const isGheUrlValid = isValidUrl(gheServerURL);
 
-	if (isGheUrlValid) {
-		try {
-			const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
-
-			if (gitHubServerApps?.length) {
-				req.log.debug(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
-				res.status(200).send({ success: true, moduleKey: "github-list-apps-page" });
-			} else {
-				req.log.debug(`No existing GitHub apps found for url: ${gheServerURL}. Making request to provided url.`);
-				await axios.get(gheServerURL);
-				res.status(200).send({ success: true, moduleKey: "github-app-creation-page" });
-			}
-		} catch (err) {
-			req.log.debug(`Something went wrong: ${gheServerURL}`, err);
-			// TODO - adding error mapping (create confluence page with error codes)
-			res.status(200).send({ success: false, error: "Something went wrong!!!!" });
+	if (!isGheUrlValid) {
+		const { errorCode, message, statusCode } = getGheErrorMessages("invalidUrl");
+		res.status(statusCode).send({ success: false, errorCode, message });
+		req.log.error(`The entered URL is not valid. ${gheServerURL} is not a valid url`);
+	}
+
+	try {
+		const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
+
+		if (gitHubServerApps?.length) {
+			req.log.debug(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
+			res.status(200).send({ success: true, moduleKey: "github-list-apps-page" });
```

Comment:
for clarity, is `moduleKey: "github-list-apps-page"` there to direct where the frontend needs to go next?  This would be break separation of concerns.  This api is only concerned with if the ghe url is valid, not the flow of the app which should be handled by the frontend.

Instead, I'd "generalize" the return type as something like `{valid:boolean; appExists: boolean; errors?:Error[]}`  which should give you all the information you need for the frontend to make it's own decision on how to move forward :)

Reply:
Good point on separation of concerns. Will update.